src/odemis/util/test/img_test.py

```diff
@@ -1181,8 +1185,827 @@ def test_rotate_img_metadata(self):
+        single_size = [[self.tile_collection[0][0]]]
+
+        # normal input of CTZYX dimension of DA's in tuple list of tuples
+        output_pos = img.getCenterOfTiles(single_size, self.tile_collection[0][0].shape)
+
+        x_pos = self.tile_collection[0][0].metadata[model.MD_POS][0]
+        y_pos = self.tile_collection[0][0].metadata[model.MD_POS][1]
+
+        # check if the x positions match
+        self.assertEqual(output_pos[0], x_pos)
+        # check if the y positions match
+        self.assertEqual(output_pos[1], y_pos)
+
+    def test_get_center_of_tiles_incorrect_shape(self):
+        """
+        Verify that using a wrong shaped DA tiles list will raise a IndexError exception
+        """
+        new_tiles = []
+
+        for t1 in self.tile_collection:
+            new_tile = []
+            for t2 in t1:
+                # remove the first 2 axis so effectively keep 3D ZYX array
+                new_tile.append(t2[0, 0, :, :, :])
+            new_tiles.append(new_tile)
+
+        result_shape = tuple(len(new_tiles) * elem for elem in self.size)
+
+        # using wrong shape force IndexError shape of CTZYX expected
+        with self.assertRaises(IndexError):
+            img.getCenterOfTiles(new_tiles, result_shape)
+
+    def test_get_center_of_tiles_no_pixelsize(self):
+        """
+        Verify that leaving out PIXEL_SIZE in the metadate DA tiles list will raise a ValueError exception
+        """
+        del self.tile_collection[0][0].metadata[model.MD_PIXEL_SIZE]
+        result_shape = tuple(len(self.tile_collection) * elem for elem in self.size)
+
+        # force ValueError by leaving out PIXEL_SIZE
+        with self.assertRaises(ValueError):
+            img.getCenterOfTiles(self.tile_collection, result_shape)
+
+
+class TestGetBoundingBox(unittest.TestCase):
+
+    def setUp(self):
+        # create a 5D size and metadata with necessary items
+        self.size = (9, 10, 13, 400, 600)
+        self.md = {
+            model.MD_DIMS: 'CTZYX',
+            model.MD_PIXEL_SIZE: (1e-6, 1e-6),
+            model.MD_POS: (0.95, 0.07)
+        }
+
+    def test_get_bounding_box_normal(self):
+        """
+        Verify that using a 5D shaped DA will return the right bounding box.
+        normal use of function
+        """
+        # create a DA with random values
+        new_da = numpy.random.random(self.size)
+        new_da = model.DataArray(new_da, self.md)
+
+        output_pos = img.getBoundingBox(new_da)
+
+        # set the positions to check for
+        pos_x = new_da.metadata[model.MD_POS][0]
+        pos_y = new_da.metadata[model.MD_POS][1]
+
+        # TODO Explain why it can be almost equal
+        # check if all the positions are almost equal
+        self.assertAlmostEqual(output_pos[0], pos_x, places=2)
+        self.assertAlmostEqual(output_pos[1], pos_y, places=2)
+        self.assertAlmostEqual(output_pos[2], pos_x, places=2)
+        self.assertAlmostEqual(output_pos[3], pos_y, places=2)
```

Comment:
this test is not checking the correct thing. The function returns a bounding box and pos_x and pos_y are the centers of the DA and thus of the bounding box. You want to check the following:

```python
expected_pos_x_min = pos_x - self.size[-1] * self.md[model.MD_PIXEL_SIZE]
self.assertEqual(output_pos[0], expected_pos_x_min)

expected_pos_x_max = pos_x + self.size[-1] * self.md[model.MD_PIXEL_SIZE]
self.assertEqual(output_pos[2], expected_pos_x_max)
```

and do the same for y. Then you probably don't need the `assertAlmostEqual`, although there might be a floating point error.